File: 1876-substrings-of-size-three-with-distinct-characters/1876-substrings-of-size-three-with-distinct-characters.go
func countGoodSubstrings(s string) int {
	var answer int
	for i := 0; i <= len(s)-3; i++ {
		if isGoodString(s[i : i+3]) {
			answer++
		}
	}
	return answer
}

func isGoodString(s string) bool {
	hashMap := make(map[rune]bool)
	for _, v := range s {
		if _, ok := hashMap[v]; ok {
			return false
		}
		hashMap[v] = true
	}
	return true
}

File: 1876-substrings-of-size-three-with-distinct-characters/1876-substrings-of-size-three-with-distinct-characters.java
import java.util.*;

class Solution {
    public int countGoodSubstrings(String s) {
        int answer = 0;
        for (int i = 0; i <= s.length()-3; i++) {
            if (isGoodString(s.substring(i,i+3))) {
                answer++;
            }
        }
        return answer;
    }

    static public boolean isGoodString(String s) {
        HashMap<Character,Boolean> hashMap = new HashMap<>();
        for (char ch : s.toCharArray()) {
            if (hashMap.containsKey(ch)) {
                return false;
            } else {
                hashMap.put(ch, true);
            }
        }
        return true;
    }
}